solutions/gold/cf-919D.mdx

```diff
@@ -105,4 +179,59 @@ public class Substring {
 
 </JavaSection>
 
+<PySection>
+```py
+from collections import deque
+
+n, m = map(int, input().split())
+a = input()
+
+adj = [[] for _ in range(n)]
+in_degree = [0] * n
+for _ in range(m):
+	x, y = map(int, input().split())
+	x -= 1
+	y -= 1
+	in_degree[y] += 1
+	adj[x].append(y)
+
+queue = deque()
+# dp[i][j] is the frequency of letter j when we are at node i.
+dp = [[0] * 26 for _ in range(n)]
+
+for i in range(n):
+	if in_degree[i] == 0:
+		queue.append(i)
+		dp[i][ord(a[i]) - ord('a')] += 1
+
+# topological sort.
+size = 0
+while size < n and len(queue) > 0:
+	cur = queue.popleft()
+	for nxt in adj[cur]:
+		for j in range(26):
+			# update the frequency with the next occurrence.
+			if j == ord(a[nxt]) - ord('a'):
+				dp[nxt][j] = max(dp[cur][j] + 1, dp[nxt][j])
+			else:
+				dp[nxt][j] = max(dp[cur][j], dp[nxt][j])
+		# add the next node to the queue.
+		in_degree[nxt] -= 1
+		if in_degree[nxt] == 0:
+			queue.append(nxt)
+	size += 1
+
+# no answer.
+if size < n:
+	print(-1)
+# find the maximum frequency across all nodes and letters.
+else:
```

Comment:
```suggestion
else:  # find the maximum frequency across all nodes and letters.
```
also if you're gonna use periods might as well go all the way & capitalize sentences asw